terraform/modules/application_secrets/kms.tf

```diff
@@ -0,0 +1,52 @@
+resource "aws_kms_key" "this" {
+  description         = "KMS key used to encrypt the Secrets Manager secrets."
+  enable_key_rotation = true
+}
+
+resource "aws_kms_alias" "this" {
+  name          = "alias/${var.prefix}-app-secretsmanager-secrets"
+  target_key_id = aws_kms_key.this.key_id
+}
+
+resource "aws_kms_key_policy" "this" {
+  key_id = aws_kms_key.this.id
+  policy = data.aws_iam_policy_document.kms.json
+}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_iam_policy_document" "kms" {
+  #checkov:skip=CKV_AWS_109:Only assigning the kms:GenerateDataKey and kms:Decrypt permissions led to a 'you won't be able to manage the key once created' error
+  #checkov:skip=CKV_AWS_111:Only assigning the kms:GenerateDataKey and kms:Decrypt permissions led to a 'you won't be able to manage the key once created' error
+  #checkov:skip=CKV_AWS_356:Only assigning the kms:GenerateDataKey and kms:Decrypt permissions led to a 'you won't be able to manage the key once created' error
+  statement {
+    principals {
+      type        = "AWS"
+      identifiers = [var.ecs_task_execution_role_arn]
+    }
+
+    actions = [
+      "kms:GenerateDataKey",
+      "kms:Decrypt"
+    ]
+
+    resources = [
+      aws_kms_key.this.arn
+    ]
+  }
+
+  statement {
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"]
+    }
+
+    actions = [
+      "kms:*"
+    ]
+
+    resources = [
+      "*"
```

Comment:
Does this need to be on everything or could it be on this specific kms key? (if so that would avoid some of the checkov skips, and the repetition, although I agree we're going to want this for every key)

Reply:
Tested this out and looks like it can just be on the specific kms key, I'll update across the codebase 👍

Reply:
Done 👍